akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/Effect.scala

```diff
@@ -20,43 +20,44 @@ object Effect {
    *
    * Side effects can be chained with `andThen`
    */
-  def persist[Event, State](event: Event): Effect[Event, State] = Persist(event)
+  def persist[Event, State](event: Event): EffectBuilder[Event, State] = Persist(event)
 
   /**
    * Persist multiple events
    *
    * Side effects can be chained with `andThen`
    */
-  def persist[Event, A <: Event, B <: Event, State](evt1: A, evt2: B, events: Event*): Effect[Event, State] =
+  def persist[Event, A <: Event, B <: Event, State](evt1: A, evt2: B, events: Event*): EffectBuilder[Event, State] =
     persist(evt1 :: evt2 :: events.toList)
 
   /**
    * Persist multiple events
    *
    * Side effects can be chained with `andThen`
    */
-  def persist[Event, State](events: im.Seq[Event]): Effect[Event, State] =
+  def persist[Event, State](events: im.Seq[Event]): EffectBuilder[Event, State] =
```

Comment:
do you think `EffectBuilder` is ok as name? my thinking is that it's something that you can add more to

Reply:
I think it's ok, but also a bit weird since you can also return that builder not built so to speak.

Reply:
I agree it's a bit unfortunate that `EffectBuilder` gives the impression that you have to call something like `.build()` to get the actual Effect, which ofc is not the case here... I don't have a better suggestion though. `OpenEffect`?